### ISSUE TITLE
Board Manager not showing esp32 to install 

### ISSUE THREAD

**Issue by usmanshahid001** [espressif/arduino-esp32#3505]
hi i updated this link:  https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json. in preferences of Arduino IDE but now esp32 is not showing in list when i open board manager.however i tried this link :  https://raw.githubusercontent.com/jantje/arduino-esp32/master/package/package_Espressif_esp32_index.json and esp32 is showing but only version 1.01

**Comment by chegewara** [comment]
Try this one:
https://dl.espressif.com/dl/package_esp32_dev_index.json

**Comment by me-no-dev** [comment]
don't ... we are trying to move away from dl.espressif.com. the json is hosted on github, if that is down.... the package is down as well

**Comment by me-no-dev** [comment]
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json this link works fine for me and shows all version (1.0.4 included)

**Comment by jramonlab** [comment]
Hi Me-no-dev,

The provided link is down:
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_dev_index.json

Why is it always down ?
Is it down on purpose for any reason?
Is there any problem on using the one from dl.espressif.com ?

Regards, Ramon.

**Comment by atanisoft** [comment]
try this: https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json

**Comment by me-no-dev** [comment]
have no idea what to tell you.. link is working fine here. It's hosted on github. Maybe check your internet? What happens when you try to open it in a browser?

**Comment by atanisoft** [comment]
@me-no-dev the above link is a better link to use and will work more reliably since it is cached via CDN.

**Comment by jramonlab** [comment]
> have no idea what to tell you.. link is working fine here. It's hosted on github. Maybe check your internet? What happens when you try to open it in a browser?

It does not load on a browser.

**Comment by jramonlab** [comment]
> @jramonlab try this: https://espressif.github.io/arduino-esp32/package_esp32_dev_index.json
> or: https://espressif.github.io/arduino-esp32/package_esp32_index.json

The first link is loaded ok in the browser, the second one (dev) is not .

**Comment by atanisoft** [comment]
> The first link is loaded ok in the browser, the second one (dev) is not .

In that case use the one of those that does work. It is the same as the raw link that @me-no-dev suggested *BUT* it sits behind a CDN with cache. Very likely your system is unable to reach the raw link due to a device blocking connections to it somewhere between your system and the GitHub backend servers.

**Comment by jramonlab** [comment]
Hi atanisoft, thanks for the info and the link :)

**Comment by tyeth** [comment]
Can we promote the boards manager link to *not use* the raw content url, and to use the github pages version instead to benefit from CDN and avoid throttling. My arduino ide v2 is always stalling getting the esp32 package list using the raw url. I've switched now to the github.io version and it flys along with no delay/pausing.

TLDR: (update docs page https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html#installing-using-arduino-ide to use github.io pages link instead of rawcontent.github.com)

**Comment by VojtechBartoska** [comment]
Hello @tyeth,

Thanks for your comment.

what do you think @pedrominatel?

**Comment by Sahil-Kashyap-in** [comment]
Guys, I'm new to IoT can anyone help me with this? 
I'm actually trying to install the esp32 board manager on the Arduino IDE software with all these links what am I supposed to do to get the esp32 installed on my PC?

**Comment by tyeth** [comment]
@Sahil-Kashyap-in 
Best place to start is the docs!
Here's the specific page to add the esp32 board support packages to the arduino IDE for your reference:
https://espressif-docs.readthedocs-hosted.com/projects/arduino-esp32/en/latest/installing.html

**Comment by Sahil-Kashyap-in** [comment]
> @Sahil-Kashyap-in Best place to start is the docs! Here's the specific page to add the esp32 board support packages to the arduino IDE for your reference: https://espressif-docs.readthedocs-hosted.com/projects/arduino-esp32/en/latest/installing.html

Thank you @tyeth I actually figured it out by adding the links on the Arduino IDE, btw thanks for the documentation link !
